File: bench/wordfreq/wordfreq.py
# Word frequency counting: string split, dict updates, and a sort with a key.
vocab = ["alpha", "beta", "gamma", "delta", "epsilon", "zeta", "eta"]

words = []
v = 0
while v < 7:
    n = 0
    while n <= v:
        words.append(vocab[v])
        n = n + 1
    v = v + 1
line = " ".join(words)

counts = {}
round_ = 0
while round_ < 150000:
    parts = line.split(" ")
    i = 0
    while i < len(parts):
        w = parts[i]
        counts[w] = counts.get(w, 0) + 1
        i = i + 1
    round_ = round_ + 1

items = list(counts.items())
items.sort(key=lambda item: item[1], reverse=True)
j = 0
while j < len(items):
    print(items[j][0], items[j][1])
    j = j + 1

File: bench/wordfreq/wordfreq.go
// Word frequency counting: string split, map updates, and a sort with a key.
package main

import (
	"fmt"
	"sort"
	"strings"
)

func main() {
	vocab := []string{"alpha", "beta", "gamma", "delta", "epsilon", "zeta", "eta"}

	words := []string{}
	v := 0
	for v < 7 {
		n := 0
		for n <= v {
			words = append(words, vocab[v])
			n = n + 1
		}
		v = v + 1
	}
	line := strings.Join(words, " ")

	counts := map[string]int{}
	round := 0
	for round < 150000 {
		parts := strings.Split(line, " ")
		i := 0
		for i < len(parts) {
			w := parts[i]
			counts[w] = counts[w] + 1
			i = i + 1
		}
		round = round + 1
	}

	type item struct {
		word  string
		count int
	}
	items := []item{}
	for word, count := range counts {
		items = append(items, item{word, count})
	}
	sort.Slice(items, func(i, j int) bool { return items[i].count > items[j].count })
	j := 0
	for j < len(items) {
		fmt.Println(items[j].word, items[j].count)
		j = j + 1
	}
}
